lib/peer-resolver/src/client.rs

```diff
@@ -0,0 +1,154 @@
+use anyhow::Result;
+use libp2p::{
+    futures::StreamExt,
+    identity::Keypair,
+    rendezvous::{client, Cookie, Namespace},
+    swarm::{NetworkBehaviour, SwarmEvent},
+    {noise, tcp, yamux, Multiaddr, PeerId, Swarm},
+};
+
+use crate::error::ResolverError;
+
+/// This struct holds the rendezvous client behaviour for registration and discovery.
+#[derive(NetworkBehaviour)]
+pub struct ClientBehaviour {
+    rendezvous: client::Behaviour,
+}
+
+/// Rendezvous client swarm
+pub struct ClientSwarm {
```

Comment:
`ClientSwarm` is super generic
```suggestion
pub struct RendezvousClientSwarm {
```